src/core/backbone_router/bbr_manager.cpp

```diff
@@ -118,6 +126,99 @@ void Manager::SendMulticastListenerRegistrationResponse(const Coap::Message &
     otLogInfoBbr("Sent MLR.rsp (status=%d): %s", aStatus, otThreadErrorToString(error));
 }
 
+void Manager::HandleDuaRegistration(const Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo)
+{
+    otError                    error               = OT_ERROR_NONE;
+    ThreadStatusTlv::DuaStatus status              = ThreadStatusTlv::kDuaSuccess;
+    bool                       isPrimary           = Get<BackboneRouter::Local>().IsPrimary();
+    uint32_t                   lastTransactionTime = 0;
+    Ip6::Address               target;
+    Ip6::InterfaceIdentifier   meshLocalIid;
+
+    VerifyOrExit(aMessage.IsConfirmable() && aMessage.GetCode() == OT_COAP_CODE_POST, error = OT_ERROR_PARSE);
+
+    SuccessOrExit(error = Tlv::FindTlv(aMessage, ThreadTlv::kTarget, &target, sizeof(target)));
+    SuccessOrExit(error = Tlv::FindTlv(aMessage, ThreadTlv::kMeshLocalEid, &meshLocalIid, sizeof(meshLocalIid)));
+
+#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
+    if (mDuaResponseIsSpecified)
+    {
+        if (mDuaResponseTargetMlIid.IsUnspecified() || mDuaResponseTargetMlIid == meshLocalIid)
+        {
+            mDuaResponseIsSpecified = false;
+            ExitNow(status = mDuaResponseStatus);
+        }
+    }
+#endif
+
+    VerifyOrExit(isPrimary, status = ThreadStatusTlv::kDuaNotPrimary);
+    VerifyOrExit(Get<BackboneRouter::Leader>().IsDomainUnicast(target), status = ThreadStatusTlv::kDuaInvalid);
+
+    if (Tlv::FindUint32Tlv(aMessage, ThreadTlv::kLastTransactionTime, lastTransactionTime) == OT_ERROR_NONE)
+    {
+        // TODO: (DUA) hasLastTransactionTime = true;
+    }
+
+    // TODO: (DUA) Add ND-PROXY table management
+    // TODO: (DUA) Add DAD process
+    // TODO: (DUA) Extended Address Query
+
+exit:
+
+    otLogInfoBbr("Received DUA.req on %s: %s", isPrimary ? "PBBR" : "SBBR", otThreadErrorToString(error));
+
+    if (error == OT_ERROR_NONE)
+    {
+        SendDuaRegistrationResponse(aMessage, aMessageInfo, target, status);
+    }
+}
+
+void Manager::SendDuaRegistrationResponse(const Coap::Message &      aMessage,
+                                          const Ip6::MessageInfo &   aMessageInfo,
+                                          const Ip6::Address &       aTarget,
+                                          ThreadStatusTlv::DuaStatus aStatus)
+{
+    otError        error   = OT_ERROR_NONE;
+    Coap::Message *message = nullptr;
+
+    VerifyOrExit((message = Get<Coap::Coap>().NewMessage()) != nullptr, error = OT_ERROR_NO_BUFS);
+
+    SuccessOrExit(message->SetDefaultResponseHeader(aMessage));
+    SuccessOrExit(message->SetPayloadMarker());
+
+    SuccessOrExit(Tlv::AppendUint8Tlv(*message, ThreadTlv::kStatus, static_cast<uint8_t>(aStatus)));
```

Comment:
do we need the `static_cast<uint8_t>(aStatus)`? I think if we explicitly define the `DuaStatus` enum as `uint8_t` it would work without requiring the cast...